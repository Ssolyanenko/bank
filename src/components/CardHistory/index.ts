export * from './History';
export * from './Pagination';
export * from './CardHistory';
