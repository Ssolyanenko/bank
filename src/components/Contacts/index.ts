export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
