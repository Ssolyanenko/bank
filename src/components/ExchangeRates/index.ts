export * from './ExchangeRates';
export * from './Converter';
export * from './Rates';
