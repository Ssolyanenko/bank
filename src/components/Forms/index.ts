export * from './Login';
export * from './PasswordRecoveryForm';
export * from './GeneralInfo';
export * from './Security';
export * from './OrderDebitTravelCard';
export * from './SetPinCodeForm';
export * from './OrderCreditCardPremium';
export * from './FormNotifications';
export * from './FilterHistoryForm';
export * from './CardLimitForm';
export * from './OrderCreditCardBilliable';
export * from './OrderDebitSmartCard';
export * from './TransferByPhoneNumberForm';
export * from './AcceptTerms';
