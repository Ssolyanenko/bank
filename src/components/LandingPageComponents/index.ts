export * from './LandingPageHeader';
export * from './LandingPageMain';
export * from './LandingPageFooter';
