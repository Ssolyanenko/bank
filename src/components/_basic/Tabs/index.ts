export * from './Tabs';
export * from './TabTitle';
