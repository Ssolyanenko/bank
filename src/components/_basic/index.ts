export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
