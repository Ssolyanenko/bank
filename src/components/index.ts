export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
