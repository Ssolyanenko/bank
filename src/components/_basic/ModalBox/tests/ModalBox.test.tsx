import 'jsdom-global/register';
import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';

import { ModalBox } from 'components/_basic';
import { mockStore } from 'helpers';
import { mockInitialState } from 'constants/mockInitialState';
import { ButtonNames } from 'constants/text';
import * as hooks from 'hooks';

jest.mock('hooks', () => {
  return {
    __esModule: true,
    ...jest.requireActual('hooks'),
  };
});

describe('<ModalBox />', (): void => {
  test('should make a snapshot', (): void => {
    const store = mockStore(mockInitialState);
    const props = {
      onSend: jest.fn(),
      messageTimeOut: <div>Error</div>,
      timeBlock: '',
      setIsActiveNewRequest: jest.fn(),
      isActiveNewRequest: false,
    };
    const wrapper = shallow(
      <Provider store={store}>
        <ModalBox {...props} />
      </Provider>
    );
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should call dispatch on click', (): void => {
    jest.useFakeTimers();
    const setTimeoutSpy = jest.spyOn(global, 'setTimeout');

    const dispatchMock = jest.fn();
    const useTypedDispatch = jest.spyOn(hooks, 'useTypedDispatch');
    useTypedDispatch.mockImplementation(() => dispatchMock);

    const store = mockStore(mockInitialState);

    const props = {
      onSend: jest.fn(),
      messageTimeOut: <div>Error</div>,
      setIsActiveNewRequest: jest.fn(),
      timeBlock: '',
      isActiveNewRequest: false,
    };
    const wrapper = mount(
      <Provider store={store}>
        <ModalBox {...props} />
      </Provider>
    );

    wrapper.find(`button[children="${ButtonNames.SEND_AGAIN}"]`).simulate('click');
    expect(dispatchMock).toBeCalledTimes(1);

    jest.clearAllTimers();
    setTimeoutSpy.mockReset();
    setTimeoutSpy.mockRestore();
  });
});
