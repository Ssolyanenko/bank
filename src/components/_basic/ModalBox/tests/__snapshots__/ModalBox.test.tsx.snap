// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBox /> should make a snapshot 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ModalBox
    isActiveNewRequest={false}
    messageTimeOut={
      <div>
        Error
      </div>
    }
    onSend={[MockFunction]}
    setIsActiveNewRequest={[MockFunction]}
    timeBlock=""
  />
</Context.Provider>
`;
